exporter/exporterparser/prometheus.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporterparser
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/census-instrumentation/opencensus-service/data"
+	"github.com/census-instrumentation/opencensus-service/exporter"
+	"github.com/orijtech/prometheus-go-metrics-exporter"
+	prometheus_golang "github.com/prometheus/client_golang/prometheus"
+)
+
+type prometheusConfig struct {
+	// Namespace if set, exports metrics under the provided value.
+	Namespace string `yaml:"namespace"`
+
+	// ConstLabels are values that are applied for every exported metric.
+	ConstLabels prometheus_golang.Labels `yaml:"const_labels"`
+
+	// The address on which the Prometheus scrape handler will be run on.
+	Address string `yaml:"address"`
+}
+
+var errBlankPrometheusAddress = errors.New("expecting a non-blank address to run the Prometheus metrics handler")
+
+// PrometheusExportersFromYAML parses the yaml bytes and returns exporter.MetricsExporters
+// targeting Prometheus according to the configuration settings.
+// It allows HTTP clients to scrape it on endpoint path "/metrics".
+func PrometheusExportersFromYAML(config []byte) (tes []exporter.TraceExporter, mes []exporter.MetricsExporter, doneFns []func() error, err error) {
```

Comment:
Is `tes` needed?

Reply:
Yes, it is needed because when parsing we use a generic function signature and multiplex on the names i.e.
<img width="607" alt="screen shot 2019-01-11 at 11 53 49 am" src="https://user-images.githubusercontent.com/4898263/51023274-983eac80-1597-11e9-84b0-155be5e8620a.png">